### PR TITLE
Add DiagnosticQuickFixesProvider

### PR DESCRIPTION
**Pull request by rita-sonarsource**

### CODE REVIEW

File: omnisharp-dotnet/src/Services/DiagnosticWorker/QuickFixes/DiagnosticQuickFixesProvider.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            {
+                return Array.Empty<IQuickFix>();
+            }
+
+            var codeFixActions = await diagnosticCodeActionsProvider.GetCodeActions(diagnostic, document);
+            var quickFixes = new List<IQuickFix>();
+
+            foreach (var action in codeFixActions)
+            {
+                var fixes = new List<Fix>();
```
Comment (rita-sonarsource) [inline omnisharp-dotnet/src/Services/DiagnosticWorker/QuickFixes/DiagnosticQuickFixesProvider.cs:100]: Everything in the `foreach` loop is not tested yet

**Comment by rita-sonarsource** [comment]
@duncanp-sonar please review, I will merge PR https://github.com/SonarSource/sonarlint-omnisharp/pull/141 into this one so the coverage will be higher + product code of the Operations will match what we discussed. I think it might be easier to review this before I merge.

File: omnisharp-dotnet/src/Services.UnitTests/DiagnosticWorker/QuickFixes/DiagnosticQuickFixesProviderTests.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            Func<Task> act = async () => await testSubject.Handle(null);
+
+            await act.Should().ThrowExactlyAsync<NotSupportedException>();
+        }
+
+        [TestMethod]
+        public async Task GetDiagnosticQuickFixes_DocumentDoesNotExist_EmptyList()
+        {
+            var diagnostic = CreateDiagnostic();
+            var workspace = CreateOmniSharpWorkspace();
```
Comment (duncanp-sonar) [inline omnisharp-dotnet/src/Services.UnitTests/DiagnosticWorker/QuickFixes/DiagnosticQuickFixesProviderTests.cs:54]: This test is currently more a test of what happens with an empty workspace. Would it be better to create a workspace with a non-matching document, rather than a workspace with no documents?

**Review by duncanp-sonar — APPROVED** [review]
Just one minor suggestion about a test.
